controllers/storagecluster/uninstall_reconciler.go

```diff
@@ -334,7 +334,6 @@ func (r *StorageClusterReconciler) deleteResources(sc *ocsv1.StorageCluster) (re
 		&ocsStorageQuota{},
 		&ocsStorageClass{},
 		&ocsCephCluster{},
-		&ocsClusterClaim{},
```

Comment:
You need to replace this line with your new CM. 
The CM's `ensureDeleted` logic needs to run whenever a storagecluster is getting deleted and this is the way to ensure that.